# **Data Cleaning and Manipulation using Python library**

In [ ]:
#import the libraries

import pandas as pd
import numpy as np
import re
import ast
import matplotlib.pyplot as plt

In [ ]:
# Reading the TSV file into a DataFrame
data = pd.read_csv("/content/chipotle.tsv", sep='\t')

In [ ]:
df = data.copy()
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

In [ ]:
df.shape
df.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

# Missing values

In [ ]:
# Checking for missing values

missing_values = df.isnull().sum()
missing_values

order_id                 0
quantity                 0
item_name                0
choice_description    1246
item_price               0
dtype: int64

In [ ]:
# Filling missing values

df['choice_description'].fillna("None", inplace=True)
df.head(8)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   

                                  choice_description item_price  
0                                               None     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                               None     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                               None     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75

# Data Types

In [ ]:
df.dtypes

order_id               int64
quantity               int64
item_name             object
choice_description    object
item_price            object
dtype: object

In [ ]:
# Converting the item_price to float data type

df['item_price'] = df['item_price'].str.replace('$', '').astype(float)

<ipython-input-8-ecf26b774fe5>:3: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df['item_price'] = df['item_price'].str.replace('$', '').astype(float)


# Duplicate Entries

In [ ]:
# To count rows and columns

df.shape

(4622, 5)

In [ ]:
# Dropping the duplicates

df.drop_duplicates(subset=['order_id', 'item_name' , 'choice_description'], keep=False, inplace=True)
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  
0                                                  None        2.39  
1                                          [Clementine]        3.39  
2                                               [Apple]        3.39  
3                                                  None        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
...                                                 ...         ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75  

[4505 rows x 5 columns]

In [ ]:
# number of rows left, after removing the duplicates
df.shape

(4505, 5)

Number of rows left:
4622 - 117(duplicates)= 4505

# Quantity and Item price

In [ ]:
# Statistics summary

(df['quantity'].describe())
(df['item_price'].describe())

count    4505.000000
mean        7.514337
std         4.243410
min         1.090000
25%         3.990000
50%         8.750000
75%         9.390000
max        44.250000
Name: item_price, dtype: float64

Based on these statistics, the data seems consistent.

# Choice Description

In [ ]:
df['choice_description'].head(10)

0                                                 None
1                                         [Clementine]
2                                              [Apple]
3                                                 None
4    [Tomatillo-Red Chili Salsa (Hot), [Black Beans...
5    [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...
6                                                 None
7    [Tomatillo Red Chili Salsa, [Fajita Vegetables...
8    [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...
9    [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...
Name: choice_description, dtype: object

In [ ]:
#Splitting the Choice_Description column
split_columns = df['choice_description'].str.split(",", expand=True)


split_columns.columns = ['Choice_Description_1', 'Choice_Description_2', 'Choice_Description_3',
                         'Choice_Description_4', 'Choice_Description_5', 'Choice_Description_6',
                         'Choice_Description_7', 'Choice_Description_8', 'Choice_Description_9',
                         'Choice_Description_10']


df = pd.concat([df, split_columns], axis=1)

In [ ]:
#dropping the column

df.drop('choice_description', axis=1, inplace=True)

# Handling special characters

In [ ]:
columns_to_check = ['item_name', 'Choice_Description_1','Choice_Description_2','Choice_Description_3','Choice_Description_4','Choice_Description_5','Choice_Description_6','Choice_Description_7','Choice_Description_8','Choice_Description_9','Choice_Description_10']

In [ ]:
# Regular expression to identity special characters

special_chars = re.compile(r'[^a-zA-Z0-9\s]')

In [ ]:
# Apply handling to specified columns using applymap and lambda

df[columns_to_check] = df[columns_to_check].applymap(lambda x: special_chars.sub('', x) if isinstance(x, str) else x)

In [ ]:
df.head(8)

order_id  quantity                             item_name  item_price  \
0         1         1          Chips and Fresh Tomato Salsa        2.39   
1         1         1                                  Izze        3.39   
2         1         1                      Nantucket Nectar        3.39   
3         1         1  Chips and TomatilloGreen Chili Salsa        2.39   
4         2         2                          Chicken Bowl       16.98   
5         3         1                          Chicken Bowl       10.98   
6         3         1                         Side of Chips        1.69   
7         4         1                         Steak Burrito       11.75   

           Choice_Description_1 Choice_Description_2 Choice_Description_3  \
0                          None                 None                 None   
1                    Clementine                 None                 None   
2                         Apple                 None                 None   
3                          None                 None                 None   
4  TomatilloRed Chili Salsa Hot          Black Beans                 Rice   
5       Fresh Tomato Salsa Mild                 Rice               Cheese   
6                          None                 None                 None   
7     Tomatillo Red Chili Salsa    Fajita Vegetables          Black Beans   

  Choice_Description_4 Choice_Description_5 Choice_Description_6  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4               Cheese           Sour Cream                 None   
5           Sour Cream            Guacamole              Lettuce   
6                 None                 None                 None   
7          Pinto Beans               Cheese           Sour Cream   

  Choice_Description_7 Choice_Description_8 Choice_Description_9  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4                 None                 None                 None   
5                 None                 None                 None   
6                 None                 None                 None   
7            Guacamole              Lettuce                 None   

  Choice_Description_10  
0                  None  
1                  None  
2                  None  
3                  None  
4                  None  
5                  None  
6                  None  
7                  None

In [ ]:
# Checking for special characters in specified columns using str.contains()

special_chars_check = df[columns_to_check].apply(lambda col: col.str.contains(special_chars))
special_chars_check
df.head()

order_id  quantity                             item_name  item_price  \
0         1         1          Chips and Fresh Tomato Salsa        2.39   
1         1         1                                  Izze        3.39   
2         1         1                      Nantucket Nectar        3.39   
3         1         1  Chips and TomatilloGreen Chili Salsa        2.39   
4         2         2                          Chicken Bowl       16.98   

           Choice_Description_1 Choice_Description_2 Choice_Description_3  \
0                          None                 None                 None   
1                    Clementine                 None                 None   
2                         Apple                 None                 None   
3                          None                 None                 None   
4  TomatilloRed Chili Salsa Hot          Black Beans                 Rice   

  Choice_Description_4 Choice_Description_5 Choice_Description_6  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4               Cheese           Sour Cream                 None   

  Choice_Description_7 Choice_Description_8 Choice_Description_9  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4                 None                 None                 None   

  Choice_Description_10  
0                  None  
1                  None  
2                  None  
3                  None  
4                  None

# Order ID Integrity

In [ ]:
# Checkiing for unique Order Ids

unique_orders = df['order_id'].unique()

In [ ]:
# Checking for irregularities

if len(unique_orders) == df.shape[0]:
    print("Order IDs are unique and consistent.")
else:
    print("Irregularities or duplicate Order IDs.")

Irregularities or duplicate Order IDs.


In [ ]:
# Checking for missing values

missing_values = df['order_id'].isnull().sum()
if missing_values > 0:
    print(f"There are {missing_values} missing IDs.")
else:
    print("No missing IDs.")

No missing IDs.


In [ ]:
# Checking for irregularities in Order IDs

order_id_counts = df['order_id'].value_counts()
print("Counts of Order ID:")
order_id_counts

Counts of Order ID:


926     20
1483    14
1786    11
691     11
759     11
        ..
1334     1
1336     1
1339     1
59       1
1004     1
Name: order_id, Length: 1827, dtype: int64

In [ ]:
# Checking for Integrity

df[df['order_id'] == 759]

order_id  quantity                     item_name  item_price  \
1872       759         1                  Chicken Bowl       11.25   
1873       759         1                 Carnitas Bowl        9.25   
1874       759         1            Chicken Soft Tacos        8.75   
1875       759         1                   Veggie Bowl       11.25   
1876       759         1                  Chicken Bowl        8.75   
1877       759         1                  Chicken Bowl       11.25   
1878       759         2             Canned Soft Drink        2.50   
1879       759         2             Canned Soft Drink        2.50   
1880       759         4                 Bottled Water        6.00   
1881       759         2           Chips and Guacamole        8.90   
1882       759         2  Chips and Fresh Tomato Salsa        5.90   

             Choice_Description_1 Choice_Description_2 Choice_Description_3  \
1872           Fresh Tomato Salsa                 Rice          Black Beans   
1873           Fresh Tomato Salsa                 Rice          Pinto Beans   
1874           Fresh Tomato Salsa               Cheese              Lettuce   
1875     Roasted Chili Corn Salsa                 Rice          Black Beans   
1876  Tomatillo Green Chili Salsa                 Rice          Black Beans   
1877           Fresh Tomato Salsa    Fajita Vegetables                 Rice   
1878                         Coke                 None                 None   
1879                    Diet Coke                 None                 None   
1880                         None                 None                 None   
1881                         None                 None                 None   
1882                         None                 None                 None   

     Choice_Description_4 Choice_Description_5 Choice_Description_6  \
1872          Pinto Beans            Guacamole              Lettuce   
1873               Cheese              Lettuce                 None   
1874                 None                 None                 None   
1875               Cheese           Sour Cream            Guacamole   
1876               Cheese           Sour Cream              Lettuce   
1877            Guacamole              Lettuce                 None   
1878                 None                 None                 None   
1879                 None                 None                 None   
1880                 None                 None                 None   
1881                 None                 None                 None   
1882                 None                 None                 None   

     Choice_Description_7 Choice_Description_8 Choice_Description_9  \
1872                 None                 None                 None   
1873                 None                 None                 None   
1874                 None                 None                 None   
1875              Lettuce                 None                 None   
1876                 None                 None                 None   
1877                 None                 None                 None   
1878                 None                 None                 None   
1879                 None                 None                 None   
1880                 None                 None                 None   
1881                 None                 None                 None   
1882                 None                 None                 None   

     Choice_Description_10  
1872                  None  
1873                  None  
1874                  None  
1875                  None  
1876                  None  
1877                  None  
1878                  None  
1879                  None  
1880                  None  
1881                  None  
1882                  None

# Item name standardization

In [ ]:
# Checking for unique values

df['item_name']
unique_item_names = df['item_name'].unique()
unique_item_names

array(['Chips and Fresh Tomato Salsa', 'Izze', 'Nantucket Nectar',
       'Chips and TomatilloGreen Chili Salsa', 'Chicken Bowl',
       'Side of Chips', 'Steak Burrito', 'Steak Soft Tacos',
       'Chips and Guacamole', 'Chicken Crispy Tacos',
       'Chicken Soft Tacos', 'Chicken Burrito', 'Canned Soda',
       'Barbacoa Burrito', 'Carnitas Burrito', 'Carnitas Bowl',
       'Bottled Water', 'Chips and Tomatillo Green Chili Salsa',
       'Barbacoa Bowl', 'Chips', 'Chicken Salad Bowl', 'Steak Bowl',
       'Barbacoa Soft Tacos', 'Veggie Burrito', 'Veggie Bowl',
       'Steak Crispy Tacos', 'Chips and Tomatillo Red Chili Salsa',
       'Barbacoa Crispy Tacos', 'Veggie Salad Bowl',
       'Chips and Roasted ChiliCorn Salsa',
       'Chips and Roasted Chili Corn Salsa', 'Carnitas Soft Tacos',
       'Chicken Salad', 'Canned Soft Drink', 'Steak Salad Bowl',
       '6 Pack Soft Drink', 'Chips and TomatilloRed Chili Salsa', 'Bowl',
       'Burrito', 'Crispy Tacos', 'Carnitas Crispy Tacos', 

In [ ]:
pattern = r'[^a-zA-Z0-9\s]'
df['item_name']

0               Chips and Fresh Tomato Salsa
1                                       Izze
2                           Nantucket Nectar
3       Chips and TomatilloGreen Chili Salsa
4                               Chicken Bowl
                        ...                 
4617                           Steak Burrito
4618                           Steak Burrito
4619                      Chicken Salad Bowl
4620                      Chicken Salad Bowl
4621                      Chicken Salad Bowl
Name: item_name, Length: 4505, dtype: object

In [ ]:
# Removing special characters from 'Item_Name'

df['item_name']=df['item_name'].str.replace(pattern, '', regex=True)
df['item_name']

0               Chips and Fresh Tomato Salsa
1                                       Izze
2                           Nantucket Nectar
3       Chips and TomatilloGreen Chili Salsa
4                               Chicken Bowl
                        ...                 
4617                           Steak Burrito
4618                           Steak Burrito
4619                      Chicken Salad Bowl
4620                      Chicken Salad Bowl
4621                      Chicken Salad Bowl
Name: item_name, Length: 4505, dtype: object

# Relationship of Quantity an Price

In [ ]:
df[['quantity','item_price']]

quantity  item_price
0            1        2.39
1            1        3.39
2            1        3.39
3            1        2.39
4            2       16.98
...        ...         ...
4617         1       11.75
4618         1       11.75
4619         1       11.25
4620         1        8.75
4621         1        8.75

[4505 rows x 2 columns]

In [ ]:
# Statistics summary

df[['quantity', 'item_price']].describe()

quantity   item_price
count  4505.000000  4505.000000
mean      1.076138     7.514337
std       0.405580     4.243410
min       1.000000     1.090000
25%       1.000000     3.990000
50%       1.000000     8.750000
75%       1.000000     9.390000
max      15.000000    44.250000

In [ ]:
item_costs = df.groupby('item_name')['item_price'].mean().to_dict()

In [ ]:
expected_price = df['quantity'] * df['item_name'].map(item_costs)
inconsistent_prices = abs(expected_price - df['item_price']) > 0.01
df.loc[inconsistent_prices, 'item_price'] = expected_price[inconsistent_prices]

In [ ]:
df['item_price']=df['item_price'].round(2)
df['item_price']=df['item_price'].round(2)
df

order_id  quantity                             item_name  item_price  \
0            1         1          Chips and Fresh Tomato Salsa        3.29   
1            1         1                                  Izze        3.39   
2            1         1                      Nantucket Nectar        3.64   
3            1         1  Chips and TomatilloGreen Chili Salsa        2.54   
4            2         2                          Chicken Bowl       20.27   
...        ...       ...                                   ...         ...   
4617      1833         1                         Steak Burrito       10.46   
4618      1833         1                         Steak Burrito       10.46   
4619      1834         1                    Chicken Salad Bowl       11.17   
4620      1834         1                    Chicken Salad Bowl       11.17   
4621      1834         1                    Chicken Salad Bowl       11.17   

              Choice_Description_1 Choice_Description_2 Choice_Description_3  \
0                             None                 None                 None   
1                       Clementine                 None                 None   
2                            Apple                 None                 None   
3                             None                 None                 None   
4     TomatilloRed Chili Salsa Hot          Black Beans                 Rice   
...                            ...                  ...                  ...   
4617            Fresh Tomato Salsa                 Rice          Black Beans   
4618            Fresh Tomato Salsa                 Rice           Sour Cream   
4619            Fresh Tomato Salsa    Fajita Vegetables          Pinto Beans   
4620            Fresh Tomato Salsa    Fajita Vegetables              Lettuce   
4621            Fresh Tomato Salsa    Fajita Vegetables          Pinto Beans   

     Choice_Description_4 Choice_Description_5 Choice_Description_6  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                  Cheese           Sour Cream                 None   
...                   ...                  ...                  ...   
4617           Sour Cream               Cheese              Lettuce   
4618               Cheese              Lettuce            Guacamole   
4619            Guacamole              Lettuce                 None   
4620                 None                 None                 None   
4621              Lettuce                 None                 None   

     Choice_Description_7 Choice_Description_8 Choice_Description_9  \
0                    None                 None                 None   
1                    None                 None                 None   
2                    None                 None                 None   
3                    None                 None                 None   
4                    None                 None                 None   
...                   ...                  ...                  ...   
4617            Guacamole                 None                 None   
4618                 None                 None                 None   
4619                 None                 None                 None   
4620                 None                 None                 None   
4621                 None                 None                 None   

     Choice_Description_10  
0                     None  
1                     None  
2                     None  
3                     None  
4                     None  
...                    ...  
4617                  None  
4618                  None  
4619                  None  
4620                  None  
4621                  None  

[4505 rows x 14 columns]

# Top 5 items by Quantity and Price

In [ ]:
df.head()

order_id  quantity                             item_name  item_price  \
0         1         1          Chips and Fresh Tomato Salsa        3.29   
1         1         1                                  Izze        3.39   
2         1         1                      Nantucket Nectar        3.64   
3         1         1  Chips and TomatilloGreen Chili Salsa        2.54   
4         2         2                          Chicken Bowl       20.27   

           Choice_Description_1 Choice_Description_2 Choice_Description_3  \
0                          None                 None                 None   
1                    Clementine                 None                 None   
2                         Apple                 None                 None   
3                          None                 None                 None   
4  TomatilloRed Chili Salsa Hot          Black Beans                 Rice   

  Choice_Description_4 Choice_Description_5 Choice_Description_6  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4               Cheese           Sour Cream                 None   

  Choice_Description_7 Choice_Description_8 Choice_Description_9  \
0                 None                 None                 None   
1                 None                 None                 None   
2                 None                 None                 None   
3                 None                 None                 None   
4                 None                 None                 None   

  Choice_Description_10  
0                  None  
1                  None  
2                  None  
3                  None  
4                  None

# Calculate total number of orders

In [ ]:
# Calculating total order count

total_orders = df['order_id'].nunique()

print("Total Order Count:", total_orders)

Total Order Count: 1827


# Find Revenue

In [ ]:
# Calculating revenue

df['total_price'] = df['quantity'] * df['item_price']
total_revenue = df['total_price'].sum()

print("Total Revenue:", total_revenue)

Total Revenue: 40975.32


# How many different items are sold?


In [ ]:
# Calculating number of different items sold
num_different_items = df['item_name'].nunique()

print("Number of Different Items Sold:", num_different_items)

Number of Different Items Sold: 50


# Calculate Total revenue for each product

In [ ]:
# Calculating total revenue for each product
df['total_price'] = df['quantity'] * df['item_price']
revenue_per_product = df.groupby('item_name')['total_price'].sum()

print("Revenue from Each Product:")
print(revenue_per_product)

Revenue from Each Product:
item_name
6 Pack Soft Drink                         376.77
Barbacoa Bowl                             654.08
Barbacoa Burrito                          876.65
Barbacoa Crispy Tacos                     153.02
Barbacoa Salad Bowl                        87.60
Barbacoa Soft Tacos                       250.50
Bottled Water                             698.14
Bowl                                      148.00
Burrito                                    44.40
Canned Soda                               228.76
Canned Soft Drink                         680.94
Carnitas Bowl                             855.62
Carnitas Burrito                          628.08
Carnitas Crispy Tacos                     111.38
Carnitas Salad                              8.99
Carnitas Salad Bowl                        66.36
Carnitas Soft Tacos                       376.00
Chicken Bowl                             8287.02
Chicken Burrito                          6756.90
Chicken Crispy Tacos            

# Calculate Total cost and quantity for each unique item

In [ ]:
# Calculating total cost and quantity for each unique item
df['total_cost'] = df['quantity'] * df['item_price']
item_summary = df.groupby('item_name').agg({
    'quantity': 'sum',
    'total_cost': 'sum'
}).reset_index()

print("Total Cost and Quantity of Every Unique Item:")
print(item_summary)

Total Cost and Quantity of Every Unique Item:
                                item_name  quantity  total_cost
0                       6 Pack Soft Drink        55      376.77
1                           Barbacoa Bowl        64      654.08
2                        Barbacoa Burrito        89      876.65
3                   Barbacoa Crispy Tacos        12      153.02
4                     Barbacoa Salad Bowl         8       87.60
5                     Barbacoa Soft Tacos        25      250.50
6                           Bottled Water       188      698.14
7                                    Bowl         4      148.00
8                                 Burrito         6       44.40
9                             Canned Soda       122      228.76
10                      Canned Soft Drink       331      680.94
11                          Carnitas Bowl        71      855.62
12                       Carnitas Burrito        60      628.08
13                  Carnitas Crispy Tacos         8      1

# Calculate Average quantity and Average cost per unit for each unique item

In [ ]:
# Calculating average quantity and average cost per unit for each unique item
item_summary = df.groupby('item_name').agg({
    'quantity': 'mean',
    'item_price': 'mean'
}).reset_index()

print("Average Quantity and Average Cost per Unit of Every Unique Item:")
print('\n',item_summary)

Average Quantity and Average Cost per Unit of Every Unique Item:

                                 item_name  quantity  item_price
0                       6 Pack Soft Drink  1.018519    6.732407
1                           Barbacoa Bowl  1.000000   10.220000
2                        Barbacoa Burrito  1.000000    9.850000
3                   Barbacoa Crispy Tacos  1.090909   11.923636
4                     Barbacoa Salad Bowl  1.000000   10.950000
5                     Barbacoa Soft Tacos  1.000000   10.020000
6                           Bottled Water  1.287671    2.367466
7                                    Bowl  2.000000   29.600000
8                                 Burrito  1.000000    7.400000
9                             Canned Soda  1.220000    1.622600
10                      Canned Soft Drink  1.177936    1.731922
11                          Carnitas Bowl  1.044118   11.308088
12                       Carnitas Burrito  1.016949   10.301864
13                  Carnitas Crispy T